jax/_src/lax/lax.py

```diff
@@ -4681,53 +4711,92 @@ def _scatter_jvp(primals, tangents, *, update_jaxpr, update_consts,
   partial(_scatter_batching_rule, scatter))
 
 
-def _reduce_shape_rule(operand, init_value, *, computation, jaxpr, consts,
-                       dimensions):
-  return tuple(np.delete(operand.shape, dimensions))
+def _reduce_abstract_eval_rule(*avals, **params):
+  assert all(isinstance(aval, UnshapedArray) for aval in avals), avals
+  avals = tuple(raise_to_shaped(aval) for aval in avals)
+  operand_avals, init_value_avals = split_list(avals, [len(avals) // 2])
+  least_specialized = _max(
```

Comment:
It feels a bit excessive to need to write out this logic. Is there a reason we couldn't keep using `standard_abstract_eval` here?

Reply:
`standard_abstract_eval` currently only handles primitives that don't have `multiple_results=True` (it only returns a single aval).

Reply:
Perhaps it's worth doing that generalization and making `standard_abstract_eval` support such primitives? I think it's simpler to have separate shape and dtype rules where possible.